### ISSUE TITLE
Allow N(dice count) to be ignored and equal to 1 by default

### ISSUE THREAD

**Issue by hmqgg** [skizzerz/DiceRoller#5]
The Basic dice expression is NdX, if N is ignored, then roll 1dX.

This feature is commonly used and required by many users, hope we could get it :)

**Comment by hmqgg** [comment]
It looks like that I need to do some dirty hack and it'll make GlobalFunction, which used T_Identifier, limited by first letter(avoid d).

**Comment by skizzerz** [comment]
The parser would need to be modified to make the leading number optional. I’m just not sure if that will cause ambiguities in the grammar with other features of the roller as of yet. I *think* it should be fine as there are enough differences between say “d20+2” (rolling a d20 and adding 2) vs “d20(2)” (calling a global function named “d20” with 2 as its argument)

**Comment by skizzerz** [comment]
There is indeed one issue with allowing this: there is no way to tell apart a global function call named `d` versus a bare roll with a variable amount of sides.

Currently, the following is supported:
```
d(1d6) # Call a global function named "d" with the result of a 1d6 roll as its argument
1d(1d6) # Roll 1d6, then roll a die with that result's number of sides
        # (for example, if the 1d6 rolled a 4, the overall expression
        # would then subsequently roll 1d4)
```

If we allow the 1 in front to be optional, there is no way to distinguish what was meant for the above case. I'm not sure what a good resolution for this would be; I do not wish to implement arbitrary restrictions about the names of global functions.

**Comment by hmqgg** [comment]
Thanks, I think it needs a dirty hack to implement this.

**Comment by skizzerz** [comment]
I figured out a way to do this without blocking any existing functionality, so I'm working on that now.

**Comment by skizzerz** [comment]
Fixed in version 4.1.0 -- the leading number in front of "d" is now optional.

How an expression like `d(1d6)` is handled depends on a few things:
1. If the expression has functions or extras on it, such as `d(1d6)ro=1` or `d(1d6).rerollOnce(=1)`, it will be interpreted as a die roll with a variable number of sides.
2. If the expression has no functions or extras and a global function named `d` is defined, it will be interpreted as a call to the global function named `d`.
3. Otherwise, it is interpreted as a die roll with a variable number of sides.
